### PR TITLE
Removed useless methods and added rate methods

### PR DESCRIPTION
**Pull request by michelmans**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,6 +10,7 @@ buildscript {
         classpath "org.jetbrains.kotlin:kotlin-allopen:$kotlin_version"
         classpath "org.jetbrains.kotlin:kotlin-noarg:$kotlin_version"
         classpath 'com.palantir:jacoco-coverage:0.4.0'
+
```
Comment (Requinard) [inline build.gradle]: remove this newline?

File: src/main/java/domain/Rate.kt [hunk 2]
```diff
@@ -23,7 +31,7 @@ class Rate : Serializable {
     constructor() {}
 
     constructor(vehicleType: VehicleType, kmPrice: Double, vignetteType: VignetteType) {
-        this.id = UUID.randomUUID().toString()
+        //this.id = UUID.randomUUID()
```
Comment (Requinard) [inline src/main/java/domain/Rate.kt]: Why is this commented? Either use it or remove it, we can get this back from git history

File: src/main/java/domain/Rate.kt [hunk 3]
```diff
@@ -11,7 +12,14 @@ import java.util.*
 class Rate : Serializable {
 
     @Id
-    var id: String? = null
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    var id: Long? = null
+
+    /*@Id
```
Comment (Requinard) [inline src/main/java/domain/Rate.kt]: Idem on the commented

File: src/main/java/domain/Vignette.kt [hunk 4]
```diff
@@ -11,7 +12,14 @@ import java.util.*
 class Vignette : Serializable {
 
     @Id
-    var id: String ?= null
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    var id: Long? = null
+
+    /*@Id
```
Comment (Requinard) [inline src/main/java/domain/Vignette.kt]: Idem on commenting

File: src/main/java/rest/VignetteResource.kt [hunk 5]
```diff
@@ -13,6 +13,7 @@ import javax.ws.rs.core.MediaType
 import javax.ws.rs.core.Response
 
 @Path("vignettes")
+@Stateless
```
Comment (Requinard) [inline src/main/java/rest/VignetteResource.kt]: why is this now stateless?

File: src/main/java/rest/RateResource.kt [hunk 6]
```diff
@@ -12,6 +12,7 @@ import javax.ws.rs.core.MediaType
 import javax.ws.rs.core.Response
 
 @Path("rates")
+@Stateless
```
Comment (Requinard) [inline src/main/java/rest/RateResource.kt]: Why is this now stateless? Was it tested?

File: build.gradle (hunk 1 above)
Comment (michelmans) [reply build.gradle]: Fixed

File: src/main/java/domain/Rate.kt (hunk 2 above)
Comment (michelmans) [reply src/main/java/domain/Rate.kt]: Fixed

File: src/main/java/domain/Rate.kt (hunk 3 above)
Comment (michelmans) [reply src/main/java/domain/Rate.kt]: Fixed

File: src/main/java/domain/Vignette.kt (hunk 4 above)
Comment (michelmans) [reply src/main/java/domain/Vignette.kt]: Fixed

File: src/main/java/rest/VignetteResource.kt (hunk 5 above)
Comment (michelmans) [reply src/main/java/rest/VignetteResource.kt]: Fixed

File: src/main/java/rest/RateResource.kt (hunk 6 above)
Comment (michelmans) [reply src/main/java/rest/RateResource.kt]: Fixed

**Pull request merged**